src/abstractitensornetwork.jl

```diff
@@ -870,6 +870,77 @@ function ITensors.commoninds(tn1::AbstractITensorNetwork, tn2::AbstractITensorNe
   return inds
 end
 
+"""Check if an itensornetwork has multiple indices along at least one of its edges"""
+function has_multi_edge(tn::AbstractITensorNetwork)
+  for e in edges(tn)
+    if length(linkinds(tn, e)) > 1
+      return false
+    end
+  end
+
+  return true
+end
+
+"""Add two itensornetworks together by growing the bond dimension. The network structures need to be have the same vertex names, same site index on each vertex """
+function add(tn1::AbstractITensorNetwork, tn2::AbstractITensorNetwork)
+  @assert issetequal(vertices(tn1), vertices(tn2))
+
+  #Check for any edges with multiple indices and combine to avoid
+  if has_multi_edge(tn1)
+    tn1 = combine_linkinds(tn1)
+  end
+
+  if has_multi_edge(tn2)
+    tn2 = combine_linkinds(tn2)
+  end
+
+  edges_tn1, edges_tn2 = edges(tn1), edges(tn2)
+
+  if !issetequal(edges_tn1, edges_tn2)
+    new_edges = union(edges_tn1, edges_tn2)
+    tn1 = insert_missing_internal_inds(tn1, new_edges)
+    tn2 = insert_missing_internal_inds(tn2, new_edges)
+  end
+
+  edges_tn1, edges_tn2 = edges(tn1), edges(tn2)
+  @assert issetequal(edges_tn1, edges_tn2)
+
+  tn12 = copy(tn1)
+  edge_index_map = Dict{NamedEdge,Index}()
+  #Create vertices of tn12 as direct sum of tn1[v] and tn2[v]. Work out the matching indices by matching edges. Make index tags those of tn1[v]
+  for v in vertices(tn1)
+    @assert siteinds(tn1, v) == siteinds(tn2, v)
+    e1_v = NamedEdge[NamedEdge(v => vn) for vn in neighbors(tn1, v)]
+    e2_v = NamedEdge[NamedEdge(v => vn) for vn in neighbors(tn2, v)]
+
+    @assert Set(e1_v) == Set(e2_v)
+    tn1v_linkinds = Index[first(linkinds(tn1, e)) for e in e1_v]
```

Comment:
Could use `only(linkinds(tn1, e))` instead of `first(linkinds(tn1, e))` to add an assertion that the edge has only one Index.